prow/crier/reporters/gcs/testutil/testutil.go

```diff
@@ -33,35 +35,88 @@ func (ca Fca) Config() *config.Config {
 }
 
 type TestAuthor struct {
-	AlreadyUsed bool
-	Bucket      string
-	Path        string
-	Content     []byte
-	Overwrite   bool
-	Closed      bool
+	Handlers map[string]map[string]*TestFileHandler
+}
+
+func (ta *TestAuthor) GetOrNewHandler(bucket, path string, create bool) *TestFileHandler {
+	if ta.Handlers == nil {
+		ta.Handlers = make(map[string]map[string]*TestFileHandler)
+	}
+	if _, ok := ta.Handlers[bucket]; !ok {
+		ta.Handlers[bucket] = make(map[string]*TestFileHandler)
+	}
+	if _, ok := ta.Handlers[bucket][path]; !ok && create {
+		ta.Handlers[bucket][path] = &TestFileHandler{}
+	}
+	return ta.Handlers[bucket][path]
+}
+
+type TestFileHandler struct {
+	Bucket    string
+	Path      string
+	Content   []byte
+	Offset    int
+	Overwrite bool
+	Closed    bool
 }
 
 type TestAuthorWriteCloser struct {
-	author *TestAuthor
+	handler *TestFileHandler
 }
 
 func (wc *TestAuthorWriteCloser) Write(p []byte) (int, error) {
-	wc.author.Content = append(wc.author.Content, p...)
+	if len(wc.handler.Content) > 0 {
+		if !wc.handler.Overwrite {
+			return 0, fmt.Errorf("already exist: %w", pkgio.PreconditionFailedObjectAlreadyExists)
+		}
+		wc.handler.Content = make([]byte, 0)
+	}
+	wc.handler.Content = append(wc.handler.Content, p...)
 	return len(p), nil
 }
 
 func (wc *TestAuthorWriteCloser) Close() error {
-	wc.author.Closed = true
+	wc.handler.Closed = true
 	return nil
 }
 
-func (ta *TestAuthor) NewWriter(ctx context.Context, bucket, path string, overwrite bool) (io.WriteCloser, error) {
-	if ta.AlreadyUsed {
-		panic(fmt.Sprintf("NewWriter called on testAuthor twice: first for %q/%q, now for %q/%q", ta.Bucket, ta.Path, bucket, path))
+type TestAuthorReadCloser struct {
+	handler *TestFileHandler
+}
+
+func (wc *TestAuthorReadCloser) Read(p []byte) (int, error) {
+	if wc.handler == nil {
+		return 0, errors.New("not exist")
 	}
-	ta.AlreadyUsed = true
-	ta.Bucket = bucket
-	ta.Path = path
-	ta.Overwrite = overwrite
-	return &TestAuthorWriteCloser{author: ta}, nil
+	if len(p) == 0 {
+		return 0, nil
+	}
+	var count int
```

Comment:
```suggestion
	count := 0
```

Reply:
I think `var something type` is preferred way of initialization when default value is expected